document-store/src/main/java/org/hypertrace/core/documentstore/model/config/TypesafeConfigDatastoreConfigExtractor.java

```diff
@@ -40,6 +53,20 @@ private TypesafeConfigDatastoreConfigExtractor(
     this.connectionPoolConfigBuilder = ConnectionPoolConfig.builder();
     this.datastoreConfigBuilder = DatastoreConfig.builder().type(type);
     this.endpointBuilder = Endpoint.builder();
+
+    final String dataStoreType = type.type();
+    this.hostKey(dataStoreType + "." + HOST_KEY)
```

Comment:
If I'm not using one of these fields, will this cause an issue if they're missing? Basically want to make sure this is compatible if I'm not using auth, for example.

Reply:
> will this cause an issue if they're missing?

Nope. If a config path is missing, the method invocation is a no-op.